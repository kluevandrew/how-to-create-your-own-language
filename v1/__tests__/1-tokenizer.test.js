const { tokenize } = require('../1-tokenizer');

describe('tokenize', () => {
  it('should tokenize', () => {
    expect(tokenize('(print (a))')).toMatchSnapshot();
  });
});
