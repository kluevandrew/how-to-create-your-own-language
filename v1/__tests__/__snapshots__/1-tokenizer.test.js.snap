// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize should tokenize 1`] = `
Array [
  Token {
    "type": "LEFT_PAREN",
    "value": "(",
  },
  Token {
    "type": "WORD",
    "value": "print",
  },
  Token {
    "type": "WHITESPACE",
    "value": " ",
  },
  Token {
    "type": "LEFT_PAREN",
    "value": "(",
  },
  Token {
    "type": "WORD",
    "value": "a",
  },
  Token {
    "type": "RIGHT_PAREN",
    "value": ")",
  },
  Token {
    "type": "RIGHT_PAREN",
    "value": ")",
  },
  Token {
    "type": "EOF",
    "value": "",
  },
]
`;
